app/models/links.rb

```diff
@@ -0,0 +1,65 @@
+class Links
+  # Links available for indexed gems
+  LINKS = {
+    'home'     => 'homepage_uri',
+    'code'     => 'source_code_uri',
+    'docs'     => 'documentation_uri',
+    'wiki'     => 'wiki_uri',
+    'mail'     => 'mailing_list_uri',
+    'bugs'     => 'bug_tracker_uri',
+    'download' => 'download_uri'
+  }.freeze
+
+  # Links available for non-indexed gems
+  NON_INDEXED_LINKS = {
+    'docs'     => 'documentation_uri'
+  }.freeze
+
+  attr_accessor :rubygem, :version, :linkset
+
+  def initialize(rubygem, version)
+    self.rubygem = rubygem
+    self.version = version
+    self.linkset = rubygem.linkset
+  end
+
+  def links
+    version.indexed ? LINKS : NON_INDEXED_LINKS
+  end
+
+  delegate :keys, to: :links
+
+  def each
+    return enum_for(:each) unless block_given?
+    links.each do |short, long|
+      value = send(long)
+      yield short, value if value
+    end
+  end
+
+  # documentation url:
```

Comment:
Oops. Thanks. I can change to uri.

Personally, I've always liked "url" better. uri feels like geek speek. But the api uses "uri".